### DISCUSSION TITLE
Determining root view

### DISCUSSION THREAD

**Discussion by mmilesg** [davetcc/tcMenu#539] [Q&A] [answered]
Quick question I hope:

In earlier projects with TcMenu I've successfully used the following check to have a "back button" press either exit the menu (if it's showing the root), or exit editor/go up a level:

```
if (!menuMgr.getCurrentSubMenu()) { // In root menu?  
   exitMenu(); // Exit without saving  
} else { // In submenu  
   menuMgr.performDirectionMove(true); // Go back  
}
```

However, in the current version of tcMenu (4.4.0) the if condition is always false and doesn't exit the menu while displaying the root. What's the best way to handle this now? Thanks so much!

**Comment by davetcc** [comment, marked as answer]
Sorry for the delay, in a recent version we added a proper item for the root menu item on the device. The easiest backward compatible way to do that now is:

    menuMgr.getCurrentSubMenu() == &MenuManager::ROOT

**Reply by davetcc** [reply]
The background is that we're trying to tidy up loose ends, and make the APIs and device work the same way. It was realistically wrong that the device did not originally have a root menu item and led to checks for null in many places.
